File: src/08-challenge/solution.ts
export default function drawRace(indices: number[], length: number): string {
	const race: string = indices.reduce((prevValue, value, index) => {
		const reindeerI: number = (value + length) % length
		let iceTrack: string = `${'~'.repeat(reindeerI)}${'r'.slice(~reindeerI + 2)}`.padEnd(length, '~')
		return prevValue + ' '.repeat(indices.length - index - 1) + iceTrack + ` /${index + 1}\n`
	}, '')

	return race.trimEnd()
}

File: src/08-challenge/solution.py
def draw_race(indices: list[int], length: int) -> str:
    race: str = ""

    for i, element in enumerate(iterable=indices):
        reindeer_i: int = (element + length) % length
        ice_track: str = "~" * reindeer_i
        ice_track += "r" if reindeer_i else ""
        ice_track = ice_track.ljust(length, "~")

        race += " " * (len(indices) - i - 1) + ice_track + f" /{i + 1}\n"

    return race.rstrip("\n")
